src/main/java/com/redhat/devtools/intellij/tektoncd/tree/TektonTreeStructure.java

```diff
@@ -526,7 +527,8 @@ private boolean hasNewToken(NamedContext newContext, Config newConfig, NamedCont
     protected void refresh() {
         try {
             WatchHandler.get().removeAll();
-            root.load();
+            initialized.set(false);
```

Comment:
Don't understand the rationale for that as refresh semantic is to refresh the tree but does not require the cluster connection to be reset ?

Reply:
I have done this to prevent the tree from showing a wrong cluster url as root. 
When the user changes cluster, the tree is refreshed and the root is drawn again, but when this happens, the tkncli still has not been updated (the `initializeTkn` is slower) and the `tkn.getMasterUrl()` to retrieve the root name returns the old cluster url as it still points to the old cluster. So you end up having a tree that shows mixed results (values from old cluster until the tkncli update and then the correct ones).

Reply:
I saw what you said regarding `LabelAndIconDescriptor` that accept a supplier but that doesn't solve the issue i have or maybe i don't see the way to use that. When the tree is refreshed, the `tkn cli` inside my supplier function will still point to the old cluster bc its update still is not complete.

An alternative (instead of using `initialized.set(false) ....`) could be to make `root.load()` returns a completablefuture and run `mutableModelSupport.fireModified(root)` when that is complete.

Reply:
I would prefer that approach as it's more or less what we have done in IJ OpenShift and it explains better the workflow